[Composable] Add more sharding strategies to runtime test

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #106003
* #106002
* #106001
* #106000
* #105999
* #105995
* #105991
* #105207
* #105206
* __->__ #105205
* #105180
* #105176

Add more sharding strategies to ensure equivalence

Differential Revision: [D47462392](https://our.internmc.facebook.com/intern/diff/D47462392/)

rohan-varma committed:
[Composable] Add more sharding strategies to runtime test

Add more sharding strategies to ensure equivalence

Differential Revision: [D47462392](https://our.internmc.facebook.com/intern/diff/D47462392/)

[ghstack-poisoned]

awgu commented on test/distributed/_composable/fully_shard/test_fully_shard_runtime.py:
```diff
@@ -92,19 +95,24 @@ def test_training(self):
                     FSDPWrapMode.MANUAL_WRAP,
+                ],
+                "sharding_strategy": [
+                    ShardingStrategy.SHARD_GRAD_OP,
```
I think?
```suggestion
                    ShardingStrategy.FULL_SHARD,
```

awgu reviewed (approved):
LGTM. (check lint)

rohan-varma committed:
Update on "[Composable] Add more sharding strategies to runtime test"


Add more sharding strategies to ensure equivalence

Differential Revision: [D47462392](https://our.internmc.facebook.com/intern/diff/D47462392/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Composable] Add more sharding strategies to runtime test"


Add more sharding strategies to ensure equivalence

Differential Revision: [D47462392](https://our.internmc.facebook.com/intern/diff/D47462392/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Composable] Add more sharding strategies to runtime test"


Add more sharding strategies to ensure equivalence

Differential Revision: [D47462392](https://our.internmc.facebook.com/intern/diff/D47462392/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[Composable] Add more sharding strategies to runtime test"


Add more sharding strategies to ensure equivalence

Differential Revision: [D47462392](https://our.internmc.facebook.com/intern/diff/D47462392/)

[ghstack-poisoned]

rohan-varma commented:
fake tensor failure https://github.com/pytorch/pytorch/actions/runs/5663535346/job/15345654211?pr=105205 issue is a KP and fixed on master.

rohan-varma commented:
@pytorchbot merge -f "CI done"
